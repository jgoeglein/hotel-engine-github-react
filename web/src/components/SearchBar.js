import React from 'react';

class SearchBar extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = { value: '' };
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    handleSubmit(event) {
        this.props.handleSearch(this.state.value);
        event.preventDefault();
    }

    handleChange(event) {
        this.setState({value: event.target.value});
    }

    render() {
        return <form onSubmit={this.handleSubmit}>
            <div className="center-flex">
                <input type="text" value={this.state.value} onChange={this.handleChange} placeholder="Search..." />
            </div>
            <div className="center-flex">
                <input type="submit" value="Search" />
            </div>
        </form>
    }
}

export default SearchBar;