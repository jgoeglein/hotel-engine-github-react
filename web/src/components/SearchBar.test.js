import { cleanup, fireEvent, render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from './SearchBar';

afterEach(cleanup);

it('SearchBar renders without crashing', () => {
  const tree = renderer
    .create(<SearchBar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Notifies props.handleSearch correctly', () => {
    const onSearch = jest.fn();
    const { getByText, getByPlaceholderText } = render(
        <SearchBar handleSearch={onSearch} />,
    );
    fireEvent.change(getByPlaceholderText('Search...'), 'Testing 123');
    fireEvent.submit(getByText('Search'));
    expect(onSearch.mock.calls.length).toBe(1);
});
