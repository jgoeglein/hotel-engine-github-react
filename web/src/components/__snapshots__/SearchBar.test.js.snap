// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders without crashing 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="center-flex"
  >
    <input
      onChange={[Function]}
      placeholder="Search..."
      type="text"
      value=""
    />
  </div>
  <div
    className="center-flex"
  >
    <input
      type="submit"
      value="Search"
    />
  </div>
</form>
`;
